src/extlib/logger.hpp

```diff
@@ -0,0 +1,88 @@
+#ifndef CPPLOGGER_HPP
+#define CPPLOGGER_HPP
+
+#include <chrono>
+#include <iostream>
+#include <string>
+#include <fstream>
+#include <filesystem>
+#include <sstream>
+#include <iomanip>
+
+enum class LogLevel {
+    INFO,
+    WARNING,
+    ERROR,
+    DEBUG
+};
+
+enum class LogOutput {
+    CONSOLE,
+    FILE
+};
+
+class Logger {
+public:
+    static Logger* init_logging() {
```

Comment:
change to `get_instance`. The current name does not make sense since "init" is usually used to refer to an initializing task, which should get run once when a system starts. Here, we are using this function over and over again to get the logger. So `get_instance` makes more sense. Also, it's standard for a singleton class to have a function called `get_instance`.